source/table_view.tsx

```diff
@@ -35,7 +36,9 @@ export class TableView extends React.Component<Properties> implements
     for(let i = 0; i < this.props.labels.length; ++i) {
       this.headerRefs[i] = null;
     }
-    this.props.model.connect(this.forceUpdate.bind(this, null));
+    this.table = new SortedTableModel(this.props.model);
+    this.table.connect(this.forceUpdate.bind(this, null));
+    this.columnOrder = [];
```

Comment:
The ```SortedTableModel``` has a columnOrder already. Can you look to reuse it instead of maintaining a separate version of it?